### PR TITLE
Add SMT100 Soil Moisture Sensor documentation

### PR DESCRIPTION
**Pull request by piechade**
## Description:
Documentation for the new smt100 component.

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#3654

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -313,6 +313,7 @@ Environmental
     SHT3X-D, components/sensor/sht3xd, sht3xd.jpg, Temperature & Humidity
     SHT4X, components/sensor/sht4x, sht4x.jpg, Temperature & Humidity
     SHTCx, components/sensor/shtcx, shtc3.jpg, Temperature & Humidity
+    SMT100, components/sensor/smt100, smt100.jpg, Moisture &Temperature
```
Comment (jesserockz) [inline index.rst]:
```suggestion
    SMT100, components/sensor/smt100, smt100.jpg, Moisture & Temperature
```

File: components/sensor/smt100.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+
+The SMT100 sensor platform allows you to use the SMT100 soil moisture and temperature with ESPHome.
+
+.. figure:: images/smt100.jpg
+    :align: center
+    :width: 50.0%
+
+
+The communication with this integration is done over a UART bus using a simple ASCII protocol. You must therefore have a uart: entry in your configuration with both the TX and RX pins set to some pins on your board and the baud rate set to 9600.
```
Comment (jesserockz) [inline components/sensor/smt100.rst]:
```suggestion
The communication with this integration is done over a UART bus using a simple ASCII protocol. You must therefore have a ``uart:`` entry in your configuration with both the TX and RX pins set to some pins on your board and the baud rate set to 9600.
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by piechade — COMMENTED** [review]
Looks good from my side.

**Comment by piechade** [comment]
@jesserockz what's missing for a merge? Both changes requests are done.

**Comment by jesserockz** [comment]
> what's missing for a merge?

Patience =)

**Comment by piechade** [comment]
> > what's missing for a merge?
> 
> Patience =)

Ok, I was concerned about the wrong accepted request from you. I was not sure if my work is done correctly.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**